### ISSUE TITLE
PostgreSQL ODM2 blank database script and ODMTools

### ISSUE THREAD

**Issue by dbciar** [ODM2/ODM2#103]
Hello,

I'm having a problem using the ODMTools project with an instance of the ODM2 database schema in PostgreSQL.  I've posted on this issue tracker as I think it's an issue with the ODM2 blank schema rather than ODMTools - if this is not the case, apologies and I will post under ODMTools.  I believe the issue is related to a comment about removing quotation marks from the schema DDL in issue #60 as mentioned at the end of this post.

So, when I run ODMTools the database configuration window is shown, the connection details are entered, and the following error is given:

![screenshot_odm_tools_invalid_connection](https://cloud.githubusercontent.com/assets/8082568/8056088/c1f680f8-0e9c-11e5-9b8d-47adbb4d0351.png)

Looking at the database logs, a number of test queries are run such as get the current version, the current schema, the isolation level etc. and these work as expected.  The following query is the one that fails from the PostgreSQL log:

2015-06-09 10:06:05 BST [4094-27] odmtools@odmtools ERROR:  42P01: relation "Variables" does not exist at character 28
2015-06-09 10:06:05 BST [4094-28] odmtools@odmtools LOCATION:  parserOpenTable, parse_relation.c:986
2015-06-09 10:06:05 BST [4094-29] odmtools@odmtools STATEMENT:  Select "VariableCode" From "Variables" Limit 1

The output from the configuration window log is:

2015-06-09 10:54:54,341 - ERROR - odmtools.controller.frmDBConfig.validateInput() (103): (psycopg2.ProgrammingError) relation "Variables" does not exist
LINE 1: Select "VariableCode" From "Variables" Limit 1

I think the error might be related to something mentioned in the open issue #60, where the second comment mentions that the quotation marks have been removed from the PostgreSQL DDL for table/column creation.  In the script the table 'Variables' does indeed have a capital letter, and as expected all table names and columns are created with lower case names when imported into PostgreSQL.  I think this change may be the cause of the ODMTools problem.

The schema used is the one found here:
https://github.com/ODM2/ODM2/tree/master/src/blank_schema_scripts/postgresql

**Comment by horsburgh** [comment]
Hi, and thanks for posting this.   As of right now, ODMTools Python is compatible with ODM 1.1.1.  However, we will be working more on the ODM2 compatibility this summer and hope to have a version that works with ODM2 very soon.

**Comment by dbciar** [comment]
Hi, thanks for the quick reply.

It was actually your paper "Open source software for visualization and quality control of continuous hydrologic and water quality sensor data" that made me look into the project, and I went back to it this afternoon and noticed the last paragraph mentions the conversion work under way.

I think I made the assumption the fault was with the ODM due to both projects being under the same ODM2 umbrella and PostgreSQL being an option, but it's good to know that it is being developed, and I look forward to trying it once it's compatible.
